### PR TITLE
feat(api): add `to_sqlglot` method to `Schema` objects

### PR DESCRIPTION
**Pull request by cpcloud**
Consolidate code to turn Ibis schemas into sqlglot objects under a `Schema.to_sqlglot` method.

### CODE REVIEW

File: ibis/backends/sql/compilers/base.py [hunk 1]
```diff
@@ -51,15 +52,7 @@ def AlterTable(*args, kind="TABLE", **kwargs):
...
-def get_leaf_classes(op):
-    for child_class in op.__subclasses__():
-        if not child_class.__subclasses__():
-            yield child_class
-        else:
-            yield from get_leaf_classes(child_class)
-
-
-ALL_OPERATIONS = frozenset(get_leaf_classes(ops.Node))
+ALL_OPERATIONS = frozenset(get_subclasses(ops.Node))
```
Comment (jcrist) [inline ibis/backends/sql/compilers/base.py:55]: These aren't equivalent. Before we were getting only the leaf classes, now we're getting all classes (including any intermediate base classes like `Binary`). Was that intentional?

**Review by jcrist — APPROVED** [review]

File: ibis/backends/sql/compilers/base.py (hunk 1 above)
Comment (cpcloud) [reply ibis/backends/sql/compilers/base.py:55]: Hm, good point. I suppose this implementation needs to be adjusted, or I can add a flag to `get_subclasses`

Comment (jcrist) [reply ibis/backends/sql/compilers/base.py:55]: Looking into how this is used, I actually think it's better this way. We have some ops that are subclasses of other (non base-class/abstract) ops. Having some extraneous methods like `visit_Binary` seems less bad than accidentally excluding things like `Median`.

Comment (cpcloud) [reply ibis/backends/sql/compilers/base.py]: Hm, also a good point, I'll revert

**Comment by cpcloud** [comment]
Checking the clouds now.

**Comment by cpcloud** [comment]
Clouds are passing now:

```
❯ pytest -m snowflake -n 8 --dist loadgroup --snapshot-update -q && pytest -m bigquery -n auto --dist loadgroup --snapshot-update -q
bringing up nodes...
x.....x............x............................x.......x...x.......................x.................x.............xx..x.x....................x................s............xxxxxxxxxx..x.x............ [ 10%]
............x...........x..................x........x.........x.......xx...xxxxx....xxxxxx..xxx.xx.xx....x.x...x.x.x...xxxx...xxxx.xx...x.xx....x.xxxxxxxxx...x..x.....x...x..xx.xx.xx...x..xxx.....xx.. [ 20%]
...x....x..x........x..x.x....x....x.............x.x.....x......................x.....x.................x...............x........x....................................xx.....x.x..........x.....x.....x. [ 31%]
............x......x.x........................x.....x....................x.x............x.x........................x...............x.....x.......x..............x..................................x.... [ 41%]
..........sx....x...x.............................................................x........x..............................................x.xx..x....................................................... [ 52%]
..x.x.............x........x.......xx..........................xx......................xx...........x........x..........x..xx....x......................x.x............................................. [ 62%]
....x.........x...............................................................x...x...............x.............x........x.........x.....x.....x......x.x............x.......................s.......... [ 73%]
...x.....x...x.....xx....x.........x..xx..x......x....x..x.................x..x....x......x............x.......x......x.......x..........x.x.x.....x......................x.................x........... [ 83%]
................s......s...............s.xx.xx.x..x...xxxx.xxx.x..x.......x..xx....xx....s.xxx...x.............x....s.........................................s......................................... [ 94%]
...................................................................................s.........................                                                                                            [100%]
1675 passed, 10 skipped, 224 xfailed in 224.24s (0:03:44)
bringing up nodes...
x...x......x...xxxx.xxxxxxx.xxxxxxxxxx.xxxxxxxxxx.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.xxxxxxxxx.xxxxxxxxxxxxxxxxxxxxxxxxx.xxxxxxxxxxxxxxxxxxxxxxx.x.x.xx.........x.xxxxx..x.x.xxx...xx..x..x...x....sssssssss [  9%]
ssssss.sssssssssssssssxsssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss..x......x..x.x.x..x.........................x..x.x........x...........sssssssssssssssssssss................ [ 18%]
.......x..xx...............x............x.......x.x..x..x....x.....x..........x.....................................xxx......x................x.....x...............x........x..x....xx.....x........x.. [ 28%]
.....xx..x..x................x..x...x....x...........x.xx.....xxxx...x.........xxx..x....x...x..s..........x.....xx.x..................x.xx.....x..................x.......................x............ [ 37%]
.....x...........................................................................x.............x......x...................x....x........x.......x............................x......xx..xx.x..x......... [ 46%]
.........x.........x..x....x...xxx..x..x........................x....xx................x....x......x.......x........x........x...x.................x............x...........x......x..x........x.xxxxxxx [ 56%]
x....x.........x..xxxx.x.x......x.x.xxxxx.x.xx.....x...........xx..xx.xxx....Xxx........x..x......x..x.x....x......X...x...........X.xX...x.x.....x.....x........Xxx.xx...xx..x..x..x.x.........xx..xx.. [ 65%]
......x...x....s...x........x.............x..x........................s....................x.....ss...............x...x.......x.......x.....x..s.....x......x........s....x....x..x.....x......xxx.x.... [ 74%]
.x.x....x.......x..............................x...................................................s.................................................................................................... [ 84%]
.........................x.x.....x..........x........................................................................................................................................................... [ 93%]
....................................................................x.....................................................................                                                               [100%]
1662 passed, 128 skipped, 343 xfailed, 5 xpassed in 254.23s (0:04:14)
```

**Comment by cpcloud** [comment]
Going to merge this next, since it will create conflicts with some other PRs.

**Pull request merged**